### PR TITLE
Add hashmap details

### PR DESCRIPTION
**Pull request by kylegranger**
Added justification language for our usage of a hash map.
Addressed local storage issues
As for possible issues with non-perstistence, I wrote we are not aware of smoldot having any issues doing it that way, and that it would be thoroughly tested.  
I also added some subheadings, so the implementation page now has a bit of structure.

### CODE REVIEW

File: src/implementation.md [hunk 1]
```diff
@@ -67,6 +74,12 @@ To support the light client functionality we will implement the support of the f
 - /dot/sync/1
 - /dot/light/2
 
-We will use hash_map with serialization/deserialization to hold the blockchain state.
+## Database
+
+We will use a hash map to store the blockchain state.  This is an in-memory serialization/deserialization data structure, functioning as a de facto database.
+
+As local storage in a browser is severely limited (5-10 MB), we cannot have persistence of a database. Smoldot uses the same method.  Although we not aware of any issues that smoldot has with non-persistence, this aspect will be thoroughly tested.
```
Comment (mariopil) [inline src/implementation.md:81]: _Although we not aware of any issues_
I think _are_ is missing here.

**Review by mariopil — APPROVED** [review]

File: src/implementation.md [hunk 2]
```diff
@@ -67,6 +74,12 @@ To support the light client functionality we will implement the support of the f
 - /dot/sync/1
 - /dot/light/2
 
-We will use hash_map with serialization/deserialization to hold the blockchain state.
+## Database
```
Comment (GraDKh) [inline src/implementation.md:77]: Maybe "storage"?

File: src/implementation.md [hunk 3]
```diff
@@ -67,6 +74,12 @@ To support the light client functionality we will implement the support of the f
...
+
+We will use a hash map to store the blockchain state.  This is an in-memory serialization/deserialization data structure, functioning as a de facto database.
```
Comment (GraDKh) [inline src/implementation.md:79]: I would add more details: "to store current peers and block header information for some number of blocks in history"

File: src/implementation.md [hunk 4]
```diff
@@ -67,6 +74,12 @@ To support the light client functionality we will implement the support of the f
...
+
+As local storage in a browser is severely limited (5-10 MB), we cannot have persistence of a database. Smoldot uses the same method.  Although we are not aware of any issues that smoldot has with non-persistence, this aspect will be thoroughly tested.
```
Comment (GraDKh) [inline src/implementation.md:81]: Would be good to add a link to smoldot

File: src/implementation.md (hunk 2 above)
Comment (kylegranger) [reply src/implementation.md:77]: :heavy_check_mark:

File: src/implementation.md (hunk 3 above)
Comment (kylegranger) [reply src/implementation.md:79]: :heavy_check_mark:

File: src/implementation.md (hunk 4 above)
Comment (kylegranger) [reply src/implementation.md:81]: :heavy_check_mark:

**Pull request merged**